src/main/java/hashtable/HashTable.java

```diff
@@ -41,18 +48,21 @@ public void put(K key, V value) {
             node.next.prev = node;
         }
         array.set(index, node);
+        count++;
+        double loadFactor = (1.0 * count) / initialCapacity;
+        if (loadFactor > defaultLoadFactor) {
+            rehash();
+        }
     }
 
     /**
-     * Removes a pair with specified key from the table.
-     * @param key the key that needs to be removed.
-     * @exception IllegalArgumentException if a pair with specified key does not exist in table.
+     * Removes a pair with specified key from the table
+     * @param key the key that needs to be removed
+     * @throws IllegalArgumentException if a pair with specified key does not exist in table
      */
     public void remove(K key) {
         LinkedListNode<K, V> node = getNodeForKey(key);
-        if (node == null) {
-            throw new IllegalArgumentException("Value with such key was not found!");
-        }
+        assert node != null;
```

Comment:
We don't use asserts for production _logic_. Asserts are more for testing. This will throw an error, while what we want is:

```
if(node == null)
  return;
```